packages/flet/lib/src/controls/canvas.dart

```diff
@@ -541,16 +539,19 @@ Future<void> loadCanvasImage(Control shape) async {
   final completer = Completer();
   shape.properties["_loading"] = completer;
 
-  final src = shape.getString("src");
-  final srcBytes = shape.get("src_bytes") as Uint8List?;
+  final resolved = ResolvedAssetSource.from(shape.get("src"));
```

Comment:
I'd rename `resolved` to `src` and maybe wrapped `ResolvedAssetSource.from` to `getSrc`?

```dart
var src = shape.getSrc("src")
...
if (src.error != null) ...

if (src.hasUri) ...
```